### PR TITLE
add checkstyle in poms for build

### PR DESCRIPTION
**Pull request by salmma**
#### Short Description
Integrate checkstyle in build.
#### Proposed Changes
  * integrate checkstyle in poms
  * by violation get warnings instead of errors
  * copy checkstyle.xml in root

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0" encoding="UTF-8"?><!--~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (wederbn) [inline checkstyle.xml:1]: It seems that this file is a copy of the checkstyle from the docs? 
We should only keep it once and either reference the root file from the docs or use the checkstyle file in the docs for the maven plugin configuration

File: org.planqk.atlas.core/pom.xml [hunk 2]
```diff
@@ -65,47 +65,75 @@
...
+                        </dependency>
+                    </dependencies>
+                </plugin>-->
+            </plugins>
+        </build>
 
     <properties>
         <!-- Default timestamp format cannot be used for filenames on Windows -->
         <maven.build.timestamp.format>yyyy-MM-dd-hh-mm</maven.build.timestamp.format>
+        <checkstyle.config.location>../checkstyle.xml</checkstyle.config.location>
```
Comment (wederbn) [inline org.planqk.atlas.core/pom.xml:135]: Do we need the properties for all three modules?
I guess it's enough to set them in the parent pom?
At least for the versions, this should work, for the location I'm not sure but I think this should be possible too.
If not we can keep that property in the sub-poms

File: checkstyle.xml (hunk 1 above)
Comment (salmma) [reply checkstyle.xml:1]: yes, it is a copy, on my opinion we should keep it in the root like in the winery

File: org.planqk.atlas.core/pom.xml (hunk 2 above)
Comment (salmma) [reply org.planqk.atlas.core/pom.xml]: Adapted

**Pull request merged**